### ISSUE TITLE
OpenAPI component schema generation should match the field order of the Java object(s)

### ISSUE THREAD

**Issue by dhoffer** [smallrye/smallrye-open-api#2077]
In our upgrade from Swagger libraries to smallrye openapi it is not honoring the order of the fields in the entities it is generating the schema for.  So the resulting OpenAPI schema looks vastly different than the POJO it represents.   All the fields exist but in what appears to be random order which is very undesirable.

We group related fields together in the POJO and need that same order in the OpenAPI schema.   

We just discoveered that this is a regression, it works fine with Quarkus 3.16.3 but it fails when we add these later dependencies that have other fixes we need.

`                    <dependency>
                        <groupId>io.smallrye</groupId>
                        <artifactId>smallrye-open-api-core</artifactId>
                        <version>4.0.3</version>
                    </dependency>
                    <dependency>
                        <groupId>io.smallrye</groupId>
                        <artifactId>smallrye-open-api-jaxrs</artifactId>
                        <version>4.0.3</version>
                    </dependency>
                    <dependency>
                        <groupId>org.eclipse.microprofile.openapi</groupId>
                        <artifactId>microprofile-openapi-api</artifactId>
                        <version>4.0.2</version>
                    </dependency>`

**Comment by MikeEdgar** [comment]
The option `mp.openapi.extensions.smallrye.sorted-properties.enable=false` disables sorting of bean properties, but `false` is the default anyway.

Can you give an example/reproducer for this? The properties are taken from the class file in the order they were written there by the compiler. Perhaps this may be due to a mix of fields and methods used to derive the properties, or something with inheritance. If you give an example I can check for sure.

**Comment by dhoffer** [comment]
Sure I will attach a reproducer of this.  View the schema of the Group object vs the Pojo.  Again this only happens with the additional dependencies above.

[json-view-schema-example.zip](https://github.com/user-attachments/files/17837018/json-view-schema-example.zip)

**Comment by MikeEdgar** [comment]
They seem to have the same order - what am I missing?

POJO (just fields in same order from attachment):
```java
class Group {
    private UUID id;
    private Integer test1;
    private String name;
    private String description;
    private String roleId;
    private List<Role> roles;
}
```
OpenAPI:
```yaml
components:
  schemas:
    Group:
      type: object
      properties:
        id:
          $ref: "#/components/schemas/UUID"
        test1:
          type: integer
          format: int32
          description: test1 field
          examples:
          - 5
        name:
          type: string
        description:
          type: string
        roleId:
          type: string
        roles:
          type: array
          items:
            type: object
            properties:
              id:
                $ref: "#/components/schemas/UUID"
              name:
                type: string
      required:
      - test1
```

**Comment by dhoffer** [comment]
Hum that is not what I get, we get this:

`    Group:
      type: object
      required:
      - test1
      properties:
        roleId:
          type: string
        roles:
          type: array
          items:
            type: object
            properties:
              name:
                type: string
              id:
                $ref: "#/components/schemas/UUID"
        name:
          type: string
        description:
          type: string
        id:
          $ref: "#/components/schemas/UUID"
        test1:
          type: integer
          format: int32
          examples:
          - 5
          description: test1 field`

**Comment by MikeEdgar** [comment]
How are you generating that second example? The openapi.yaml in the ZIP you attached has the schema I posted in my last comment.

**Comment by MikeEdgar** [comment]
I found the issue and the fix is in #2079
